### PR TITLE
chore(github): Update actions' versions

### PR DESCRIPTION
**Pull request by yu-croco**
Signed-off-by: yu-croco <yu.croco@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -18,9 +18,9 @@ jobs:
         uses: azure/setup-helm@v1
 
       - name: Set up python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v4
```
Comment (yu-croco) [inline .github/workflows/lint-and-test.yml:21]: There is no breaking changes. :eyes:
https://github.com/actions/setup-python/releases/tag/v4.0.0

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ jobs:
           fi
 
       - name: Create kind cluster
-        uses: helm/kind-action@v1.2.0
+        uses: helm/kind-action@v1.4.0
```
Comment (yu-croco) [inline .github/workflows/lint-and-test.yml:58]: There is no breaking changes. :eyes:
https://github.com/helm/kind-action/releases/tag/v1.4.0

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -20,7 +20,10 @@ jobs:
           fetch-depth: 0
 
       - name: Install Helm
-        uses: azure/setup-helm@v1
+        uses: azure/setup-helm@v3
+        with:
+          version: latest # stable
+          token: ${{ secrets.GITHUB_TOKEN }} # only needed if version is 'latest'
```
Comment (yu-croco) [inline .github/workflows/publish.yml:26]: I set config by referring sample. 👀 
https://github.com/Azure/setup-helm#example

Screen Shot 2022-10-08 at 23 24 35


Current CI also uses latest Helm version, v3.10.0.
https://github.com/argoproj/argo-helm/actions/runs/3210246553/jobs/5247553641
Screen Shot 2022-10-08 at 23 26 25

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ jobs:
       pull-requests: write  # for actions/stale to close stale PRs
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/stale@v5
+    - uses: actions/stale@v6
```
Comment (yu-croco) [inline .github/workflows/stale.yml:17]: https://github.com/actions/stale/releases/tag/v6.0.0

> Issues/PRs default `close-issue-reason` is now `not_planned`

**Review by jmeridth — APPROVED** [review]

**Pull request merged**